servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java

```diff
@@ -557,14 +557,33 @@ protected final void initNext(CharSequence cookieHeaderValue) {
          */
         private HttpCookiePair findNext(CharSequence cookieHeaderValue) {
             int semiIndex = nextCookieDelimiter(cookieHeaderValue, nextNextStart);
-            HttpCookiePair next = DefaultHttpCookiePair.parseCookiePair(cookieHeaderValue, nextNextStart, semiIndex);
+            int nameLength = indexOf(cookieHeaderValue, '=', nextNextStart) - nextNextStart;
+            if (nameLength < 0) {
+                throw new IllegalArgumentException("no cookie value found after index " + nextNextStart +
+                        (next == null ? " (there was no previous cookie)" :
+                                " (found after parsing the '" + next.name() + "' cookie)"));
+            }
+            HttpCookiePair next = DefaultHttpCookiePair.parseCookiePair(
+                    cookieHeaderValue, nextNextStart, nameLength, semiIndex);
             if (semiIndex > 0) {
                 if (cookieHeaderValue.length() - 2 <= semiIndex) {
-                    advanceCookieHeaderValue();
-                    nextNextStart = 0;
-                } else {
-                    // skip 2 characters "; " (see https://tools.ietf.org/html/rfc6265#section-4.2.1)
+                    if (cookieParsingStrictRfc6265()) {
+                        throw new IllegalArgumentException("cookie '" + next.name() +
+                                "': cookie is not allowed to end with ;");
```

Comment:
`H2PriorKnowledgeFeatureParityTest.cookiesRoundTrip` has a `endsWithSemi` variable now which isn't picked up as invalid. can we make the parsing/validation consistent?